### PR TITLE
1259 Address Management - Bulk un-invalidate cases

### PR DESCRIPTION
**Pull request by liamtoozer**
# Motivation and Context

We need to mark cases as `addressInvalid` false from a file of case IDs sent from Address Management.

# What has changed

* Added new message end point to 'uninvalidate' (revalidate..?) cases
* Rejigged field helper to validate inbound message

# How to test?

Run against other branches and ensure ATs pass

# Links

https://trello.com/c/bbi2PO7z/1259-am-uninvalidate-cases-8

### CODE REVIEW

**Comment by liamtoozer** [comment]
[AM - uninvalidate cases (8)](https://trello.com/c/bbi2PO7z/1259-am-uninvalidate-cases-8)

File: src/main/java/uk/gov/ons/census/casesvc/model/dto/RmUnInvalidateAddress.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.ons.census.casesvc.model.dto;
+
+import java.util.UUID;
+import lombok.Data;
+
+@Data
+public class RmUnInvalidateAddress {
```
Comment (AdamHawtin) [inline src/main/java/uk/gov/ons/census/casesvc/model/dto/RmUnInvalidateAddress.java:7]: I mentioned in the toolbox PR too, I wonder if this should be `RmUninvalidateAddress`?

**Review by AdamHawtin — APPROVED** [review]
passing AT's locally 👍

**Review by LukeLoze1 — APPROVED** [review]

File: src/main/java/uk/gov/ons/census/casesvc/utility/FieldworkHelper.java [hunk 2]
```diff
@@ -3,17 +3,46 @@
...
+    if (hasRegionNorthernIrelandAndCECaseType(caze)) {
+      return false;
+    }
+
+    if (caze.getRefusalReceived() != null) {
+      return false;
+    }
+
+    if (caze.getEstabType() != null) {
+      return !caze.getEstabType().equals("TRANSIENT PERSONS");
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/utility/FieldworkHelper.java:36]: so we _do_ want to send these to `MIGRANT WORKERS`? That seems oddly specific

File: src/main/java/uk/gov/ons/census/casesvc/utility/FieldworkHelper.java [hunk 3]
```diff
@@ -3,17 +3,46 @@
...
     }
+
     if (caze.getEstabType().equals("TRANSIENT PERSONS")
         || caze.getEstabType().equals("MIGRANT WORKERS")) {
       return false;
     }
     return true;
   }
+
+  public static boolean shouldSendUnInvalidatedAddressCaseToField(Case caze, String eventChannel) {
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/utility/FieldworkHelper.java:22]: I'm not sure we can justify the proliferation of near-identical methods, with different names. I'm sure Dan said that the business logic for sending CREATE and UPDATE messages is identical. Only CANCEL has different logic.

**Review by NickGrantONS — DISMISSED** [review]
I don't want the `FieldworkHelper` to proliferate with a million different subtle variations. We've been told the business rules are identical for all CREATE and UPDATE messages to FWMT-G.

File: src/main/java/uk/gov/ons/census/casesvc/utility/FieldworkHelper.java (hunk 3 above)
Comment (liamtoozer) [reply src/main/java/uk/gov/ons/census/casesvc/utility/FieldworkHelper.java:22]: We will ask Dan about the acceptance criteria as it does seem odd that the UPDATE is similar to CREATE 👍

File: src/main/java/uk/gov/ons/census/casesvc/utility/FieldworkHelper.java (hunk 2 above)
Comment (liamtoozer) [reply src/main/java/uk/gov/ons/census/casesvc/utility/FieldworkHelper.java:36]: Yeah, it does seem odd that the acceptance criteria is different here. Will double-check with Dan 👍

**Comment by liamtoozer** [comment]
> I don't want the `FieldworkHelper` to proliferate with a million different subtle variations. We've been told the business rules are identical for all CREATE and UPDATE messages to FWMT-G.

Spoke to Dan and looks like there will be a ticket coming through shortly to consolidate the 'send to fieldwork' logic. The acceptance criteria to send the CREATE to field is a little different on this ticket, which is what we followed. Instead, we'll use the existing method in this ticket as the criteria needs to be the same 👍

**Pull request merged**